### PR TITLE
DWR-360, deal with large reconciliation report; switch to FTP

### PR DESCRIPTION
**Pull request by malaffoon**
There are changes in sbac-config-repo that will have to happen as well. Probably i'll just completely disable this task until ETS gives us FTP server credentials, and then i'll only enable it in the awsets environment.

### CODE REVIEW

File: task-service/src/main/java/org/opentestsystem/rdw/ingest/tasking/repository/JdbcReportRepository.java [hunk 1]
```diff
@@ -36,7 +31,7 @@
 @Repository
 class JdbcReportRepository implements ReportRepository {
     private static final Logger logger = LoggerFactory.getLogger(JdbcReportRepository.class);
-    private static final String[] DefaultHeaders = new String[]{"student_id", "asmt_guid", "date_taken"};
+    private static final String[] DefaultHeaders = new String[]{"student_id", "asmt_guid", "school_year", "opp_id", "batch", "import_id"};
```
Comment (malaffoon) [inline task-service/src/main/java/org/opentestsystem/rdw/ingest/tasking/repository/JdbcReportRepository.java]: part of the change was to change output columns per ETS request

File: task-service/src/main/java/org/opentestsystem/rdw/ingest/tasking/task/ReconciliationReportConfiguration.java [hunk 2]
```diff
@@ -1,42 +1,106 @@
...
+ * </ul>
+ * To enable FTP, the following properties must be defined:<ul>
+ *     <li>task.send-reconciliation-report.ftp-sender.server</li>
+ *     <li>task.send-reconciliation-report.ftp-sender.username</li>
+ *     <li>task.send-reconciliation-report.ftp-sender.password</li>
+ * </ul>
+ * To properly disable this task, remove task.send-reconciliation-report.
  */
 @Configuration
+@ConditionalOnProperty(prefix = "task.send-reconciliation-report", name = { "cron", "query" })
```
Comment (malaffoon) [inline task-service/src/main/java/org/opentestsystem/rdw/ingest/tasking/task/ReconciliationReportConfiguration.java]: I've changed this to be like the organizations task: if it isn't configured to be active, none of the beans are instantiated.

File: task-service/src/main/java/org/opentestsystem/rdw/ingest/tasking/task/ReconciliationReportConfiguration.java [hunk 3]
```diff
@@ -1,42 +1,106 @@
...
     }
 
+    @Bean
+    public ReconciliationService reconciliationService(final ReportRepository reportRepository) {
+        return new DefaultReconciliationService(reportRepository);
+    }
+
+    /**
+     * There needs to be exactly one {@link ReportSender} depending on the configuration.
+     * Instead of mucking about with @ConditionalXxx annotations, just get the properties
```
Comment (malaffoon) [inline task-service/src/main/java/org/opentestsystem/rdw/ingest/tasking/task/ReconciliationReportConfiguration.java]: @wtritch this is what my question was about.

Comment (wtritch) [inline task-service/src/main/java/org/opentestsystem/rdw/ingest/tasking/task/ReconciliationReportConfiguration.java]: Ahhh, this work.  In TDS I created an "implementation" property for switching the caching implementation between redis, guava, ehcache.

**Pull request merged**